pkg/machine/hyperv/config.go

```diff
@@ -172,6 +160,44 @@ func (v HyperVVirtualization) NewMachine(opts machine.InitOptions) (machine.VM,
 	return v.LoadVMByName(opts.Name)
 }
 
+// acquireVMImage determines if the image is already in a FCOS stream. If so,
+// retrieves the image path of the uncompressed file. Otherwise, the user has
+// provided an alternative image, so we set the image path and download the image.
+func (v HyperVVirtualization) acquireVMImage(opts machine.InitOptions) (*machine.VMFile, string, error) {
```

Comment:
looks like this is copy paste from the qemu implementation, can we move this to a common function and share it between both backend to prevent duplication?

Reply:
we can and will but it needs to be part of a larger refactor.  can we merge this and it will get done later.

Reply:
> it will get done later.

I hear this all the time but only a few people actually take such commitment seriously